### PR TITLE
Plugin point for pixels which get data stripped

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205520159940704/f 

### Description
Introduces a plugin point for defining pixels which should have their ATB and app version removed. These pixels are already defined, but defining them via the plugin point approach allows for more flexibility in _where_ the pixels are defined, allowing them to be created in other modules but still contribute towards this list which the `AtbAndAppVersionPixelRemovalInterceptor` will use.

### Steps to test this PR

- [ ] Take one of the pixels in the list and trigger the scenario where it will be used
  - e.g., `AppPixelName.EMAIL_USE_ADDRESS`
   - log into Email Protectio 
   - visit https://fill.dev/form/registration-email
   - tap on the email field
   - choose to use the personal duck address
   - verify in the logs or with a breakpoint that this is picked up as a special case and ATP/app version are removed by `AtbAndAppVersionPixelRemovalInterceptor` (e.g., add `Timber.d("Removing ATB and app version from pixel $pixel")` at line 49)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/global/api/AtbAndAppVersionPixelRemovalInterceptor.kt
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/global/api/AtbAndAppVersionPixelRemovalInterceptor.kt:72]: The plan is that most of these pixels can be moved to another module. However, is also dependent on another branch so will move them after this PR lands.

File: app/src/main/java/com/duckduckgo/app/global/api/AtbAndAppVersionPixelRemovalInterceptor.kt
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/global/api/AtbAndAppVersionPixelRemovalInterceptor.kt:41]: nit: `distinct()` converts to a set and then back to a list. The second (to list) conversion is not really necessary. We can just `toSet()` instead

**Review by aitorvs — APPROVED** [review]
Left a nit. LGTM

**Comment by CDRussell** [comment]
Thanks for the review @aitorvs. Adding you as assignee.

I've changed to using `toSet()` too. Will merge once CI checks pass.

**Pull request merged**
